File: typespec/hub/employers.ts
export interface FilterEmployersRequest {
  prefix: string;
}

export interface HubEmployer {
  domain: string;
  name: string;
  ascii_name: string;
}

export interface FilterEmployersResponse {
  employers: HubEmployer[];
}

export interface GetEmployerDetailsRequest {
  domain: string;
}

export interface HubEmployerDetails {
  name: string;
  verified_employees_count: number;
  is_onboarded: boolean;
  active_openings_count: number;
  is_following: boolean;
}

File: typespec/hub/employers.go
package hub

type FilterEmployersRequest struct {
	Prefix string `query:"prefix"`
}

type HubEmployer struct {
	Domain    string `json:"domain"`
	Name      string `json:"name"`
	AsciiName string `json:"ascii_name"`
}

type FilterEmployersResponse struct {
	Employers []HubEmployer `json:"employers"`
}

type GetEmployerDetailsRequest struct {
	Domain string `json:"domain" validate:"required,validate_domain"`
}

type HubEmployerDetails struct {
	Name                   string `json:"name"`
	VerifiedEmployeesCount uint32 `json:"verified_employees_count"`

	// Anything below this line should be ignored if IsOnboarded is false
	IsOnboarded         bool   `json:"is_onboarded"`
	ActiveOpeningsCount uint32 `json:"active_openings_count"`
	IsFollowing         bool   `json:"is_following"`
}
